### ISSUE TITLE
Null coalescing operator detected as arithmetic operation

### ISSUE THREAD

**Issue by dmolineus** [squizlabs/PHP_CodeSniffer#2325]
The null coalescing operator is detected as arithmetic operation inside the `Squiz.Formatting.OperatorBracket` sniff. The tenary operator `?:` isn't. This leads to different code style forcements. 

```php
$baz = $foo['bar'] ?? null; // Triggers the `Arithmetic operation must be bracketed`
$baz = $foo['bar'] ?: null; // No error is triggered
```

I expect that both statements are handled equally.

**Comment by gsherwood** [comment]
That sniff just reports: `Operation must be bracketed (Squiz.Formatting.OperatorBracket.MissingBrackets)`. It doesn't consider `??` an arithmetic operator.

The reason why `?:` is not reported is because it's not actually a PHP operator. It's just a standard inline if/else without the middle part of the ternary operator left blank.

**Comment by southskies** [comment]
Is this intended function or a bug? Is there a bracketing method that'd work?

Thanks for PHP_CodeSniffer, it's solved so many problems.

**Comment by gsherwood** [comment]
> Is this intended function or a bug?

The sniff is intended to work this way. It only looks for PHP operators. The `?` in a ternary statement is an inline then, the the `:` is an inline else. PHP doesn't actually assign these any special meaning, as it does for comparison operators.

> Is there a bracketing method that'd work?

There is no included sniff to enforce brackets in a way that makes `?:` look like an operator.

**Comment by dmolineus** [comment]
Thanks @gsherwood for clearifying it. Technical it makes sense, however from the code style perspective it's a bit irritating. 

> That sniff just reports: Operation must be bracketed 

At least in version 2.x the error message is: `Arithmetic operation must be bracketed`. Haven't tested it on v3 yet.

> There is no included sniff to enforce brackets in a way that makes ?: look like an operator.

Would you accept a PR for it?

**Comment by gsherwood** [comment]
> Would you accept a PR for it?

Sure, although you'd probably need another sniff to enforce no space between `?` and `:` as well. Because it's not an operator, PHP doesn't care if you have spaces, comments, newlines etc between those two tokens, but you might.

**Comment by southskies** [comment]
>>Is there a bracketing method that'd work?
>
>There is no included sniff to enforce brackets in a way that makes ?: look like an operator.

Thanks for getting back to me. 

I was referring to the Null coalescing operator, "??", not the ternary operator. Thoughts?

**Comment by gsherwood** [comment]
> I was referring to the null coalescing operator, "??", not the ternary operator.

Oh sorry - I thought you want it the other way around.

There is no way to make this sniff ignore the `??` operator with this sniff.
